### PR TITLE
nixos/mailman: increase uwsgi buffer size

### PR DESCRIPTION
**Pull request by yu-re-ka**
otherwise results in 502 bad gateway errors with some clients (which send a lot of cookies?)

Change-Id: I9aadedb7acde0388f060dbb82ccd8788f41ff0e6

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Is it a problem if we set this to big? I have set it to just 8k bit micro managing this is probably just a waste of time.

**Comment by SuperSandro2000** [comment]
> which send a lot of cookies?

derp, I answered that question in my head but didn't write it down.
It is caused by to big cookies like the ones set by oauth2-proxy in one of the setups I manage (https://gitea.c3d2.de/c3d2/nix-config/commit/3d6b1cfd069faf53b96228bddd21914449e87507). I tested it a bit 4 months ago and I think doubling the default of 4096 felt reasonable. Unless the cookies are gigantic (which I hope they are not) this should be good to go for many usecases.



> Honestly I have no idea how to find a sane value for this.

I think it is trial and error. Setting it to big might open DOS possibilities or make them easier to exploit?

> If you say we should just set it to 8k at first and see if that solves the problems, I'm happy to change it that.

If it solves the problem for you than we are already two and then I think this is a good, sane default.

---

TLDR: we change it to 8192 and I approve and merge?

File: nixos/modules/services/mail/mailman.nix [hunk 1]
```diff
@@ -595,6 +595,7 @@ in {
             plugins = ["python3"];
             home = webEnv;
             http = "127.0.0.1:18507";
+            buffer-size = 32768;
```
Comment (SuperSandro2000) [inline nixos/modules/services/mail/mailman.nix:598]:
```suggestion
            buffer-size = 8192;
```

**Review by SuperSandro2000 — APPROVED** [review]
